[Sub #10 · 2/4] Speaking-pill banner: smart truncation + emoji guard

> **Sub-issue di #10** — granularità 2/4

## Meta
- Status: To do
- Owner: @Leonardo-Corte
- Depends on: #10·1/4
- Estimate: S (≤2h)
- Component: iOS

## Output atteso
Il banner della pill in `.speaking` mostra una versione coerente del testo TTS. Oggi `handleResult` tronca a 100 caratteri spezzando le parole. Vogliamo: (1) trim alle ultime word boundary entro 80 char, (2) strippare emoji multi-codepoint (ActivityKit a volte li renderizza male), (3) fallback a "Speaking…" se il trimmed risulta < 2 char.

## Target files & anchors
| File | Line/Symbol | Cosa modifichi |
|---|---|---|
| `02_GIGI_APP/GIGI/GigiSmartOrchestrator.swift` | `handleResult()` line 224-227 | Sostituisci la logica `trimmed.count <= 100 ? trimmed : prefix(97) + "…"` con un nuovo helper `bannerForPill(speech:)` |
| `02_GIGI_APP/GIGI/GigiSmartOrchestrator.swift` | new func `bannerForPill(speech:)` near `splitMultipleIntents` (~line 421) | Nuova funzione static |

## Changes (passo per passo)
1. In `GigiSmartOrchestrator.swift` aggiungi (statica, vicino a `splitMultipleIntents`):
   ```swift
   static func bannerForPill(speech: String) -> String {
       let cleaned = speech
           .trimmingCharacters(in: .whitespacesAndNewlines)
           .unicodeScalars
           .filter { !$0.properties.isEmojiPresentation && !($0.value >= 0x1F300 && $0.value <= 0x1FAFF) }
           .reduce(into: "") { $0.append(Character($1)) }
       guard cleaned.count >= 2 else { return "Speaking…" }
       guard cleaned.count > 80 else { return cleaned }
       // Trim alla word boundary più vicina sotto 80
       let cap = cleaned.prefix(80)
       if let lastSpace = cap.lastIndex(of: " ") {
           return String(cap[..<lastSpace]) + "…"
       }
       return String(cap) + "…"
   }
   ```
2. In `handleResult` (line 224-227), sostituisci:
   ```swift
   let banner = trimmed.isEmpty
       ? "Fatto."
       : (trimmed.count <= 100 ? trimmed : String(trimmed.prefix(97)) + "…")
   ```
   con:
   ```swift
   let banner = trimmed.isEmpty ? "Fatto." : Self.bannerForPill(speech: trimmed)
   ```

## Build verify
```bash
ssh <USER>@<HOST> "cd ~/GIGI/02_GIGI_APP && /usr/bin/xcodebuild -project GIGI.xcodeproj -scheme GIGI -configuration Debug -destination 'generic/platform=iOS' CODE_SIGN_IDENTITY='' CODE_SIGNING_REQUIRED=NO CODE_SIGNING_ALLOWED=NO build 2>&1 | grep -E 'error:|BUILD SUCCEEDED|BUILD FAILED' | head -40"

--- a/02_GIGI_APP/GIGI/GigiSmartOrchestrator.swift
+++ b/02_GIGI_APP/GIGI/GigiSmartOrchestrator.swift
@@ -243,9 +243,7 @@ class GigiSmartOrchestrator: ObservableObject {
         }
 
         let trimmed = result.speech.trimmingCharacters(in: .whitespacesAndNewlines)
-        let banner = trimmed.isEmpty
-            ? "Fatto."
-            : (trimmed.count <= 100 ? trimmed : String(trimmed.prefix(97)) + "…")
+        let banner = trimmed.isEmpty ? "Fatto." : Self.bannerForPill(speech: trimmed)
 
         // T5: empty speech path. Skip TTS (avoids `speak("")` → empty AVSpeech buffer
         // → mDataByteSize=0 noise), close the pill straight away.
@@ -459,6 +457,25 @@ class GigiSmartOrchestrator: ObservableObject {
     }
 
     // MARK: - Helpers
+
+    /// Formats a TTS speech string into a Live Activity pill banner: trims to last
+    /// word boundary under 80 chars, strips emoji presentation scalars (ActivityKit
+    /// renders some multi-codepoint emoji incorrectly), falls back to "Speaking…"
+    /// if the cleaned text is too short.
+    static func bannerForPill(speech: String) -> String {
+        let cleaned = speech
+            .trimmingCharacters(in: .whitespacesAndNewlines)
+            .unicodeScalars
+            .filter { !$0.properties.isEmojiPresentation && !($0.value >= 0x1F300 && $0.value <= 0x1FAFF) }
+            .reduce(into: "") { $0.append(Character($1)) }
+        guard cleaned.count >= 2 else { return "Speaking…" }
+        guard cleaned.count > 80 else { return cleaned }
+        let cap = cleaned.prefix(80)
+        if let lastSpace = cap.lastIndex(of: " ") {
+            return String(cap[..<lastSpace]) + "…"
+        }
+        return String(cap) + "…"
+    }
 
     /// Splits a text containing multiple sequential commands into individual parts.
     /// Returns nil if only one command is detected (avoids false splits like "call mom and dad").

